examples/mlperf/model_eval.py

```diff
@@ -42,6 +42,57 @@ def input_fixup(x):
     print(f"****** {n}/{d}  {n*100.0/d:.2f}%")
     st = time.perf_counter()
 
+def eval_retinanet():
+  # RetinaNet with ResNeXt50_32X4D
+  from models.resnet import ResNeXt50_32X4D
+  from models.retinanet import RetinaNet
+  mdl = RetinaNet(ResNeXt50_32X4D())
+  mdl.load_from_pretrained()
+
+  input_mean = Tensor([0.485, 0.456, 0.406]).reshape(1, -1, 1, 1)
+  input_std = Tensor([0.229, 0.224, 0.225]).reshape(1, -1, 1, 1)
+  def input_fixup(x):
+    x = x.permute([0,3,1,2]) / 255.0
+    x -= input_mean
+    x /= input_std
+    return x
+
+  from datasets.openimages import openimages, iterate
+  from pycocotools.coco import COCO
+  from pycocotools.cocoeval import COCOeval
+  coco = COCO(openimages())
+  coco_eval = COCOeval(coco, iouType="bbox")
+  coco_results, evaluated_imgs = [], []
+
+  from tinygrad.jit import TinyJit
+  mdlrun = TinyJit(lambda x: mdl(input_fixup(x)).realize())
+
+  n, bs, eval_freq = 0, 8, 200
+  st = time.perf_counter()
+  for x, targets in iterate(coco, bs):
+    dat = Tensor(x.astype(np.float32))
+    mt = time.perf_counter()
+    if dat.shape[0] == bs:
+      outs = mdlrun(dat).numpy()
+    else:
+      mdlrun.jit_cache = None
+      outs =  mdl(input_fixup(dat)).numpy()
+    et = time.perf_counter()
+    predictions = mdl.postprocess_detections(outs, input_size=dat.shape[1:3], orig_image_sizes=[t["image_size"] for t in targets])
+    evaluated_imgs.extend([t["image_id"] for t in targets])
+    coco_results.extend([{"image_id": targets[i]["image_id"], "category_id": label, "bbox": box, "score": score}
+                    for i, prediction in enumerate(predictions) for box, score, label in zip(*prediction.values())])
```

Comment:
Spacing.

Also, isn't this eval O(n^2)? There's no way to do it incrementally?

Reply:
I might have found a way to do it with pycocotools, i'll test it